zebra-state/src/service/non_finalized_state.rs

```diff
@@ -182,16 +183,17 @@ impl NonFinalizedState {
     fn validate_and_commit(
         &self,
         parent_chain: Chain,
-        prepared: PreparedBlock,
+        mut prepared: PreparedBlock,
         finalized_state: &FinalizedState,
     ) -> Result<Chain, ValidateContextError> {
-        check::utxo::transparent_spend(
+        let utxos = check::utxo::transparent_spend(
             &prepared,
             &parent_chain.unspent_utxos(),
             &parent_chain.spent_utxos,
             finalized_state,
         )?;
 
+        prepared.block_utxos.extend(utxos);
```

Comment:
This might not be the best way but this is the spot where we have all the utxos we need so we extend the `block_utxos` field with them.